aws_lambda_builders/workflows/go_modules/workflow.py

```diff
@@ -0,0 +1,45 @@
+"""
+Go Modules Workflow
+"""
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+from aws_lambda_builders.actions import CopySourceAction
+
+from .actions import GoModulesBuildAction
+from .builder import GoModulesBuilder
+from .utils import OSUtils
+
+
+class GoModulesWorkflow(BaseWorkflow):
+
+    NAME = "GoModulesBuilder"
+
+    CAPABILITY = Capability(language="go",
+                            dependency_manager="modules",
+                            application_framework=None)
+
+    def __init__(self,
+                 source_dir,
+                 artifacts_dir,
+                 scratch_dir,
+                 manifest_path,
+                 runtime=None,
+                 osutils=None,
+                 **kwargs):
+
+        super(GoModulesWorkflow, self).__init__(
+            source_dir,
+            artifacts_dir,
+            scratch_dir,
+            manifest_path,
+            runtime=runtime,
+            **kwargs)
+
+        if osutils is None:
+            osutils = OSUtils()
+
+        executable_name = osutils.basename(source_dir)
```

Comment:
In the Go Dep workflow, the executable name is defined to match the handler specified in the function. This detail should remain the same here otherwise the name of the binary could differ from what is in the template.

For reference: https://github.com/awslabs/aws-lambda-builders/pull/54/files#diff-c11c930189e8062aced3a130ff5ff0efR48

Reply:
Makes sense, fixed!